src/common/data-modeler-service/TableActions.ts

```diff
@@ -153,7 +176,28 @@ export class TableActions extends DataModelerActions {
         this.dataModelerStateService.dispatch("addOrUpdateTableToState",
             [table, isNew]);
 
-        await this.importTableDataByType(table);
+        const response = await this.importTableDataByType(table);
+        if (response?.status !== undefined && (response?.status === ActionStatus.Failure)) {
+            if (isNew) {
+                // Delete the table entirely.
+                this.dataModelerStateService.dispatch("deleteEntity",
+                [EntityType.Table, StateType.Derived, derivedTable.id]);
+                // Fetch the persistent table in this instance
+                // and delete
+                const existingTable = this.dataModelerStateService
+                    .getEntityStateService(EntityType.Table, StateType.Persistent)
+                    .getByField("tableName", table.name);
+                this.dataModelerStateService.dispatch("deleteEntity",
+                    [EntityType.Table, StateType.Persistent, existingTable.id]);
+            } else {
+                this.dataModelerStateService.dispatch("updateEntity",
+                    [EntityType.Table, StateType.Persistent, originalPersistentTable]);
+                // Reset entity status to idle in the case where the table already exists.
+                this.dataModelerStateService.dispatch("setEntityStatus",
```

Comment:
No need to do this explicitly. `@DataModelerActions.ResetStateToIdle(EntityType.Table)` takes care of this.